### PR TITLE
Reorg and refactor to be closer to current standards

### PR DESCRIPTION
**Pull request by geojaz**
No major functional changes in this one- other than to get the automated tests to pass.

I've followed the direction of the other helmsman repos in doing a minor restructuring. It's come to my attention that this impacts other projects that have this dependency. 

To that effect- I think we should restructure, whether it happens today or in the future. But it looks like we're going to need to adopt a versioning strategy and some guidelines about expectations around this. I have a couple of thoughts on this one, but I actually think we should avoid becoming held up on process right now and see where this goes. That said, I completely agree, it can be a little chaotic. 

I think one thing that we can do to help steer in the right direction is to consider what we might consider a "library" or a "package" and what we'd consider a "demo" or a "project" and to start organically evolving our process to reflect that. Ie: much stricter versioning on assets that could be attractive to use as building blocks for other projects.

Anyway, I pushed this forward before I saw @chrislovecnm's comment- and we've spoken. This is still highly WIP and I've somewhat redirected/paused on this today for other priorities.

**CHANGELOG**
* No more .sh in /. Created scripts/
* moved .tf and supporting modules to to terraform/

* moved the elasticsearch stuff into one place 
* updated checks and linting and paths and everything and got Jenkins wired up with the invaluable assistance of @jramsdale 

Side note: i'd like to move towards IAC repos that are of a broader scale than just terraform/.... we don't currently have a home for kubernetes manifests like the policy directory and as more moves to kubernetes, we need a cohesive strategy. i have many thoughts on this if you want to have a really   boring gvc about project structure, just let me know :)

### CODE REVIEW

**Comment by geojaz** [comment]
Helmsman CI checks are not expected to pass until #9 goes in.

File: Makefile [hunk 1]
```diff
@@ -90,26 +90,27 @@ bootstrap:
 # Step 2: terraform is used to automate the terraform workflow
 .PHONY: terraform
 terraform:
-	terraform fmt
-	terraform validate -check-variables=false
-	terraform init
-	terraform apply -var "project=$(PROJECT)" -auto-approve
+	terraform init terraform/
+	terraform fmt terraform/
```
Comment (chrislovecnm) [inline Makefile]: Should fmt be here?

File: scripts/cloud-create.sh [hunk 2]
```diff
@@ -21,12 +21,13 @@
...
 # "---------------------------------------------------------"
 
+set -o errexit
 set -o nounset
 set -o pipefail
 
-ROOT=$(dirname "${BASH_SOURCE[0]}")
-# shellcheck disable=SC1090
-source "$ROOT"/k8s.env
+PROJECT_ROOT=$(git rev-parse --show-toplevel)
```
Comment (chrislovecnm) [inline scripts/cloud-create.sh]: here

File: WORKSPACE [hunk 3]
```diff
@@ -11,7 +11,62 @@
...
+    name = "io_bazel_rules_k8s",
+    commit = "62ae7911ef60f91ed32fdd48a6b837287a626a80",
+    remote = "https://github.com/bazelbuild/rules_k8s.git",
+)
+
+load("@io_bazel_rules_k8s//k8s:k8s.bzl", "k8s_repositories", "k8s_defaults")
+
+k8s_repositories()
+
+_CLUSTER = "gke_pso-helmsman-cicd-infra_us-west1-b_cloud-cluster"
```
Comment (chrislovecnm) [inline WORKSPACE]: We should parameterize this. Add as a todo

**Pull request merged**